### PR TITLE
Enable users to save corrections to the ASN field in the UI

### PR DESCRIPTION
**Pull request by richrace**
Screenshot 2024-03-18 at 16 47 11

Screenshot 2024-03-18 at 16 47 28

Screenshot 2024-03-18 at 17 11 06

### CODE REVIEW

**Comment by richrace** [comment]
Checked the status of Old Bichard, and it was correctly updated.

File: cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/arrestSummonsNumber.cy.ts [hunk 1]
```diff
@@ -199,13 +241,13 @@ describe("ASN", () => {
...
     cy.get(".moj-badge").contains("Correction").should("not.exist")
     cy.get(".Defendant-details-table").contains("AAAAAAAAAAAAAAAAAAA")
+    cy.get("#asn").clear()
     cy.get("#asn").type("1101ZD0100000448754K")
 
     cy.get("button").contains("Submit exception(s)").click()
     cy.get("button").contains("Submit exception(s)").click()
 
     cy.get(".Defendant-details-table").contains("1101ZD0100000448754K")
-    cy.get(".moj-badge").contains("Correction").should("exist")
```
Comment (richrace) [inline cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/arrestSummonsNumber.cy.ts:208]: I think this should be there

File: src/features/CourtCaseDetails/Tabs/Panels/DefendantDetails.tsx [hunk 2]
```diff
@@ -87,8 +153,8 @@ export const DefendantDetails = ({ amendFn, amendmentRecords }: DefendantDetails
               }
             </HintText>
             <HintText>{"Example: 22 49AB 49 1234 C"}</HintText>
-            <div className={`${asnFormGroupError}`}>
-              {!isValidAsn && (
+            <div className={!hideError() ? `${asnFormGroupError}` : ""}>
+              {!hideError() && (
```
Comment (richrace) [inline src/features/CourtCaseDetails/Tabs/Panels/DefendantDetails.tsx:157]: Need to refactor

File: src/features/CourtCaseDetails/Tabs/Panels/DefendantDetails.tsx [hunk 3]
```diff
@@ -99,9 +165,17 @@ export const DefendantDetails = ({ amendFn, amendmentRecords }: DefendantDetails
                 name={"asn"}
                 onChange={handleAsnChange}
                 value={(amendmentRecords.asn as string) ?? ""}
-                error={!isValidAsn}
+                error={!hideError()}
```
Comment (richrace) [inline src/features/CourtCaseDetails/Tabs/Panels/DefendantDetails.tsx:168]: And again here

File: src/pages/api/court-cases/[courtCaseId]/update.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import withApiAuthentication from "middleware/withApiAuthentication/withApiAuthentication"
```
Comment (richrace) [inline src/pages/api/court-cases/[courtCaseId]/update.ts:1]: Need tests 😬

File: src/middleware/withApiAuthentication/withApiAuthentication.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import parseJwtCookie from "middleware/withAuthentication/parseJwtCookie"
```
Comment (richrace) [inline src/middleware/withApiAuthentication/withApiAuthentication.ts:1]: Need tests 😬

File: cypress/component/DefendantDetails.cy.tsx [hunk 6]
```diff
@@ -102,7 +102,7 @@ describe("Defendant Details", () => {
 
     cy.mount(
       <CourtCaseContext.Provider value={{ courtCase }}>
-        <DefendantDetails amendFn={() => () => {}} amendmentRecords={{}} />
+        <DefendantDetails amendFn={() => () => {}} amendmentRecords={{}} stopLeavingFn={() => {}} />
```
Comment (richardcane) [inline cypress/component/DefendantDetails.cy.tsx:105]: Am I wrong in thinking that we now have two standards for this? `onAmend` `onLeave` vs. above

**Review by richardcane — APPROVED** [review]

**Pull request merged**